python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py

```diff
@@ -0,0 +1,223 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+import logging
+import os
+from copy import copy
+from typing import Optional
+
+import yaml
+
+from semantic_kernel.connectors.ai.open_ai.prompt_execution_settings.open_ai_prompt_execution_settings import (
+    OpenAIChatPromptExecutionSettings,
+)
+from semantic_kernel.connectors.ai.open_ai.services.open_ai_chat_completion import OpenAIChatCompletion
+from semantic_kernel.connectors.ai.open_ai.utils import (
+    get_tool_call_object,
+)
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.exceptions.planner_exceptions import PlannerInvalidConfigurationError
+from semantic_kernel.functions.kernel_arguments import KernelArguments
+from semantic_kernel.functions.kernel_function import KernelFunction
+from semantic_kernel.kernel import Kernel
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+from semantic_kernel.planners.function_calling_stepwise_planner.function_calling_stepwise_planner_options import (
+    FunctionCallingStepwisePlannerOptions,
+)
+from semantic_kernel.planners.function_calling_stepwise_planner.function_calling_stepwise_planner_result import (
+    FunctionCallingStepwisePlannerResult,
+    UserInteraction,
+)
+from semantic_kernel.planners.planner_extensions import PlannerKernelExtension
+from semantic_kernel.prompt_template.kernel_prompt_template import KernelPromptTemplate
+from semantic_kernel.prompt_template.prompt_template_config import PromptTemplateConfig
+
+CUR_DIR = os.path.dirname(os.path.realpath(__file__))
+PLAN_YAML_FILE_PATH = os.path.join(CUR_DIR, "generate_plan.yaml")
+STEP_PROMPT_FILE_PATH = os.path.join(CUR_DIR, "step_prompt.txt")
+
+STEPWISE_PLANNER_PLUGIN_NAME = "StepwisePlanner_Excluded"
+
+STEPWISE_USER_MESSAGE = (
+    "Perform the next step of the plan if there is more work to do."
+    "When you have reached a final answer, use the UserInteraction-SendFinalAnswer"
+    "function to communicate this back to the user."
+)
+
+USER_INTERACTION_SEND_FINAL_ANSWER = "UserInteraction-SendFinalAnswer"
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class FunctionCallingStepwisePlanner(KernelBaseModel):
+    service_id: str
+    options: FunctionCallingStepwisePlannerOptions
```

Comment:
I notice there are also execution_settings in the Options object, what happens when those are different then the service_id here?

Reply:
I can handle this in the next iteration.